mod-audit-server/src/main/java/org/folio/dao/acquisition/impl/PieceEventsDaoImpl.java

```diff
@@ -0,0 +1,158 @@
+      Tuple queryParams = Tuple.of(UUID.fromString(pieceId), limit, offset);
+      pgClientFactory.createInstance(tenantId).selectRead(query, queryParams, promise);
+    } catch (Exception e) {
+      LOGGER.warn("Error getting order audit events by piece id: {}", pieceId, e);
+      promise.fail(e);
+    }
+    LOGGER.info("getAuditEventsByOrderId:: Retrieved AuditEvent with piece id: {}", pieceId);
+    return promise.future().map(rowSet -> rowSet.rowCount() == 0 ? new PieceAuditEventCollection().withTotalItems(0)
+      : mapRowToListOfPieceEvent(rowSet));
+  }
+
+  private PieceAuditEventCollection mapRowToListOfPieceEvent(RowSet<Row> rowSet) {
+    LOGGER.debug("mapRowToListOfOrderEvent:: Mapping row to List of Piece Events");
+    PieceAuditEventCollection pieceAuditEventCollection = new PieceAuditEventCollection();
+    rowSet.iterator().forEachRemaining(row -> {
+      pieceAuditEventCollection.getPieceAuditEvents().add(mapRowToPieceEvent(row));
+      pieceAuditEventCollection.setTotalItems(row.getInteger(TOTAL_RECORDS_FIELD));
+    });
+    LOGGER.info("mapRowToListOfOrderEvent:: Mapped row to List of Piece Events");
+    return pieceAuditEventCollection;
+  }
+
+  private PieceAuditEvent mapRowToPieceEvent(Row row) {
+    LOGGER.debug("mapRowToPieceEvent:: Mapping row to Order Event");
+    return new PieceAuditEvent()
+      .withId(row.getValue(ID_FIELD).toString())
+      .withAction(row.get(PieceAuditEvent.Action.class, ACTION_FIELD))
+      .withPieceId(row.getValue(PIECE_ID_FIELD).toString())
+      .withUserId(row.getValue(USER_ID_FIELD).toString())
+      .withEventDate(Date.from(row.getLocalDateTime(EVENT_DATE_FIELD).toInstant(ZoneOffset.UTC)))
+      .withActionDate(Date.from(row.getLocalDateTime(ACTION_DATE_FIELD).toInstant(ZoneOffset.UTC)))
+      .withPieceSnapshot(JsonObject.mapFrom(row.getValue(MODIFIED_CONTENT_FIELD)));
+  }
+
+  private void makeSaveCall(Promise<RowSet<Row>> promise, String query, PieceAuditEvent pieceAuditEvent, String tenantId) {
+    LOGGER.debug("makeSaveCall:: Making save call with query : {} and tenant id : {}", query, tenantId);
+    try {
+      pgClientFactory.createInstance(tenantId).execute(query, Tuple.of(pieceAuditEvent.getId(),
+          pieceAuditEvent.getAction(),
+          pieceAuditEvent.getPieceId(),
+          pieceAuditEvent.getUserId(),
+          LocalDateTime.ofInstant(pieceAuditEvent.getEventDate().toInstant(), ZoneOffset.UTC),
+          LocalDateTime.ofInstant(pieceAuditEvent.getActionDate().toInstant(), ZoneOffset.UTC),
+          JsonObject.mapFrom(pieceAuditEvent.getPieceSnapshot())),
+        promise);
+    } catch (Exception e) {
+      LOGGER.error("Failed to save record with id: {} for order id: {} in to table {}",
+        pieceAuditEvent.getId(), pieceAuditEvent.getPieceId(), TABLE_NAME, e);
+      promise.fail(e);
+    }
+  }
+
+  private String formatDBTableName(String tenantId, String table) {
```

Comment:
Lets extract this method in DbUtils, because we have the same method in all 3 daos

Reply:
extracted method